File: apps/ios/ios-rust-test/UI/Transfer/TransferUiState.swift
struct TransferUiState {
    var origin = ""
    var destination = ""
    /// `String`. Siempre. El estado es el último lugar donde alguien se tienta con un número.
    var amount = ""
    var accounts: [Account] = []
    var result: TransferResult?
    /// `true` mientras corre `simulatedLatencyMs`.
    var isLoading = false
    /// Ya resuelto a texto de usuario.
    var error: String?
}

File: apps/android/app/src/main/java/dev/tohure/android_rust_test/ui/transfer/TransferUiState.kt
package dev.tohure.android_rust_test.ui.transfer

import androidx.compose.runtime.Immutable
import kotlinx.collections.immutable.ImmutableList
import kotlinx.collections.immutable.persistentListOf
import uniffi.core_financiero.Account
import uniffi.core_financiero.TransferResult

@Immutable
data class TransferUiState(
    val origin: String = "",
    val destination: String = "",
    /** `String`. Siempre. El estado es el último lugar donde alguien se tienta con un número. */
    val amount: String = "",
    val accounts: ImmutableList<Account> = persistentListOf(),
    val result: TransferResult? = null,
    /** `true` mientras corre `simulatedLatencyMs`. */
    val isLoading: Boolean = false,
    /** Ya resuelto a texto de usuario, no la excepción. */
    val error: String? = null,
)
